pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -131,10 +131,26 @@ func GetPodConditionFromList(conditions []v1.PodCondition, conditionType v1.PodC
 	return -1, nil
 }
 
+// GetPodIPs returns the pod's IP addresses of given pod
+// Addresses should be fetched from Status.PodIPs if existing valuse,
+// otherwise, from Status.PodIP
+func GetPodIPs(pod *v1.Pod) []string {
+	var podIPs []string
+	if len(pod.Status.PodIPs) > 0 {
+		for _, itemIP := range pod.Status.PodIPs {
+			podIPs = append(podIPs, itemIP.IP)
+		}
+	} else {
+		podIPs = append(podIPs, pod.Status.PodIP)
```

Comment:
This is not accurate if pod.Status.PodIP==""

Reply:
sure, I will add the assertion here for checking `pod.Status.PodIP`. Thanks @howardjohn

Reply:
done